ssm/ssm_test.go

```diff
@@ -32,9 +33,62 @@ func (m *mockSSMClient) AddTagsToResource(input *ssm.AddTagsToResourceInput) (*s
 	return out, nil
 }
 
+func (m *mockSSMClient) GetParametersByPath(input *ssm.GetParametersByPathInput) (*ssm.GetParametersByPathOutput, error) {
+	var keyA = "/codacy/test/foo"
+	var keyB = "/codacy/test/sneaky"
+	var valA = "old bar"
+	var valB = "will be deleted"
+	out := &ssm.GetParametersByPathOutput{
+		NextToken: nil,
+		Parameters: []*ssm.Parameter{
+			{
+				ARN:              new(string),
+				DataType:         new(string),
+				LastModifiedDate: &time.Time{},
+				Name:             &keyA,
+				Selector:         new(string),
+				SourceResult:     new(string),
+				Type:             new(string),
+				Value:            &valA,
+				Version:          new(int64),
+			},
+			{
+				ARN:              new(string),
+				DataType:         new(string),
+				LastModifiedDate: &time.Time{},
+				Name:             &keyB,
+				Selector:         new(string),
+				SourceResult:     new(string),
+				Type:             new(string),
+				Value:            &valB,
+				Version:          new(int64),
+			},
+		},
+	}
+
+	return out, nil
+}
+
+func (m *mockSSMClient) DeleteParameters(input *ssm.DeleteParametersInput) (*ssm.DeleteParametersOutput, error) {
+	var s = "deleted test"
+	out := &ssm.DeleteParametersOutput{
+		DeletedParameters: []*string{
+			&s,
+		},
+		InvalidParameters: nil,
+	}
+
+	return out, nil
+}
+
 func TestProcessParametersWithoutErrors(t *testing.T) {
 
 	mockSvc := &mockSSMClient{}
 
-	ProcessParameters(mockSvc, parameters, "test", true)
+	ProcessParameters(mockSvc, parameters, true)
+}
+
+func TestDeleteParametersWihoutErrors(t *testing.T) {
```

Comment:
Adding to my comment above, we should add a test to make sure that deleting `/codacy/test` implies deleting only `/codacy/test/` and that some other path, say `/codacy/test-keep` is not deleted.